src/backend/modules/url/url.service.js

```diff
@@ -87,12 +87,14 @@ class UrlServiceImp {
 
     async updateOne(urlId, urlInfo, userId) {
         const isUrlExisted = await this.repository.findById(urlId);
-        if (!isUrlExisted.url) throw new NotFoundException('urlId not found');
         if (isUrlExisted.userId === userId) {
-            const isSlugExisted = await this.repository.findBySlugExist(urlInfo.slug, urlId);
-            if (isSlugExisted) throw new DuplicateException(`Slug (${urlInfo.slug}) is already existed`);
+            const isSlugExisted = await this.repository.findBySlug(urlInfo.slug);
```

Comment:
Check if found url's slug is not equal slug in request body -> check is slug existed. This will reduce the number of requests to database